boxes/generators/can_storage.py

```diff
@@ -0,0 +1,232 @@
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERself.canHightANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from boxes import *
+import math
+
+class FrontEdge(edges.BaseEdge):
+    char = "a"
+
+    def __call__(self, length, **kw):
+        x = math.ceil( ((self.canDiameter * 0.5 + 2 * self.thickness) * math.sin(math.radians(self.angle))) / self.thickness)
+        if self.top_edge != "e":
+            self.corner(90, self.thickness)
+            self.edge(0.5 * self.canDiameter)            
+            self.corner(-90, 0.25 * self.canDiameter)
+        else:
+            self.moveTo(-self.burn, self.canDiameter + self.thickness, -90)
+            self.corner(90, 0.25 * self.canDiameter)
+            self.edge(self.thickness)
+
+        self.edge(0.5 * self.canDiameter - self.thickness)
+        self.corner(-90, 0.25 * self.canDiameter)
+        self.edge(0.5 * self.canDiameter)
+        self.corner(90, self.thickness)
+        self.edge(x * self.thickness )
+        self.corner(90, self.thickness)
+        self.edge(0.5 * self.canDiameter)
+        self.corner(-90, 0.25 * self.canDiameter)
+        self.edge(0.5 * self.canDiameter - (1 + x) * self.thickness + self.top_chute_height + self.bottom_chute_height - self.barrier_height)
+        self.corner(-90, 0.25 * self.canDiameter)
+        self.edge(0.5 * self.canDiameter)
+        self.corner(90, self.thickness)
+        self.edge(self.barrier_height)
+        self.edge(self.thickness)
+
+class CanStorage(Boxes):
+    """Storage box for round containers"""
+
+    ui_group = "Misc"
+    
+    angle = 3.0 # slope angle of the chutes in degree
+    
+    def __init__(self):
+        Boxes.__init__(self)
+
+        self.addSettingsArgs(edges.FingerJointSettings, finger=2.0, space=2.0, surroundingspaces=0.0)
+        self.addSettingsArgs(fillHolesSettings)
+
+        self.argparser.add_argument(
+            "--top_edge", action="store",
+            type=ArgparseEdgeType("efhŠ"), choices=list("efhŠ"),
+            default="Š", help="edge type for top edge")
+        self.argparser.add_argument(
+            "--bottom_edge", action="store",
+            type=ArgparseEdgeType("eEš"), choices=list("eEš"),
+            default="š", help="edge type for bottom edge")
+
+        # Add non default cli params if needed (see argparse std lib)
+        self.argparser.add_argument(
+            "--canDiameter",  action="store", type=float, default=67,
+            help="outer diameter of the cans to be stored (in mm)")
+        self.argparser.add_argument(
+            "--canHight",  action="store", type=float, default=115,
+            help="hight of the cans to be stored (in mm)")
```

Comment:
I wonder if there should be a parameter for the number of rows. But having just two as a default is fine for now.